From 2dab404aa20225b0b217cb97015b673abc78a4c4 Mon Sep 17 00:00:00 2001
From: Lasse Collin <lasse.collin@tukaani.org>
Date: Wed, 9 Sep 2026 14:14:40 +0300
Subject: [PATCH] liblzma: Make lzma_lz_decoder_init() safe to reinit after
 alloc failure

If memory allocation fails and the resulting coder state is reused,
ensure that memory allocation is attempted again. However, coders that
are initialized via lzma_next_filter_init() shouldn't be reinitialized
after failure; they should be cleaned up with lzma_next_end().

Reported-by: GitHub user christos-cantina-security (christos-spearbit)
(cherry picked from commit fe4d763d566a38ad61d4c5022520c25578a3a464)
---
 src/liblzma/lz/lz_decoder.c | 1 +
 1 file changed, 1 insertion(+)

--- a/src/liblzma/lz/lz_decoder.c
+++ b/src/liblzma/lz/lz_decoder.c
@@ -260,6 +260,7 @@ lzma_lz_decoder_init(lzma_next_coder *ne
 
 	// Allocate and initialize the dictionary.
 	if (coder->dict.size != lz_options.dict_size) {
+		coder->dict.size = 0;
 		lzma_free(coder->dict.buf, allocator);
 		coder->dict.buf
 				= lzma_alloc(lz_options.dict_size, allocator);
